overlord/configstate/configcore/log_control.go

```diff
@@ -0,0 +1,79 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+)
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core.system.printk.console-loglevel"] = true
```

Comment:
IMHO there is too many levels, perhaps something shorter and more inline with existing "core.journal.persistent" and "core.system.disable-backlight-service", e.g: "core.system.kernel-loglevel"?

(let's discuss / agree here before actual code changes).

Depending on the agreed name, I would also consider renaming the files & functions to reflect it.

Reply:
Well the question here is whether we are going over time to support more and more of sysctl params, in which case we should actually try to stick to the naming and hierarchy there somehow, or just a few, in which case we have a bit more liberty to make nice names.

Reply:
There are four values in printk denote: console_loglevel, default_message_loglevel, minimum_console_loglevel and default_console_loglevel.
Based on jdstrand's suggestion in LP: #1877060, I only added an option for console_loglevel. In order to keep flexible, I used the name with hierarchy.
core.system.kernel-loglevel? core.system.kernel-console-loglevel? or other suggested names?

Reply:
we need more levels here:

```
core.system.kernel.printk.console-loglevel
```

`kernel.printk` we get from sysctl / proc/sys itself:

```
# sysctl kernel.printk
kernel.printk = 4	4	1	7
```